### ISSUE TITLE
is cloud data flow server require kafka message bus on cloud foundry?

### ISSUE THREAD

**Issue by sandip0018** [spring-cloud/spring-cloud-dataflow#590]
We have deployed our cloud dataflow server on cloud foundry but we are facing issue in launching simple timestamp task, so question is like is there kafka require for message bus as we are facing below error.

```
org.springframework.cloud.dataflow.rest.client.DataFlowClientException: value

    at org.springframework.cloud.dataflow.rest.client.VndErrorResponseErrorHandler.handleError(VndErrorResponseErrorHandler.java:52)
    at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:641)
    at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:597)
    at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:557)
    at org.springframework.web.client.RestTemplate.postForObject(RestTemplate.java:357)
    at org.springframework.cloud.dataflow.rest.client.TaskTemplate.launch(TaskTemplate.java:114)
    at org.springframework.cloud.dataflow.shell.command.TaskCommands.launch(TaskCommands.java:125)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.springframework.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:216)
    at org.springframework.shell.core.SimpleExecutionStrategy.invoke(SimpleExecutionStrategy.java:64)
    at org.springframework.shell.core.SimpleExecutionStrategy.execute(SimpleExecutionStrategy.java:57)
    at org.springframework.shell.core.AbstractShell.executeCommand(AbstractShell.java:130)
    at org.springframework.shell.core.JLineShell.promptLoop(JLineShell.java:533)
    at org.springframework.shell.core.JLineShell.run(JLineShell.java:179)
    at java.lang.Thread.run(Thread.java:745)
```

@sabbyanandan can you please provide update on this?

**Comment by sabbyanandan** [comment]
Hi, @sandip0018: 

Thanks for trying out!

For SCDF + CF questions, please use this [dedicated repo](https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/issues). This repo contains all the core components of Spring Cloud Data Flow and it has no relevance of Cloud Foundry. I'll close this issue.

> facing issue in launching simple timestamp task

SCDF's Tasks aren't supported in Cloud Foundry, yet. The core fixes that are necessary for running Tasks in Cloud Foundry is expected to be available in PCF's 1.7.2 release and we will be building upon that.

> is there kafka require for message bus
Kafka is a requirement *only* when you're running SCDF's Local server. If you're running SCDF's CF server, you've options to select between Rabbit or Kafka binders.

**Comment by sandip0018** [comment]
Thanks for reply it help us lot.
